src/librustc_mir/transform/const_prop.rs

```diff
@@ -553,6 +553,40 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
                 }
             }
 
+            Rvalue::Cast(CastKind::Misc, op, ty) => {
+                trace!("checking Cast(Misc, {:?}, {:?})", op, ty);
+
+                if ty.is_integral() && op.ty(&self.local_decls, self.tcx).is_integral() {
+                    let value = self.use_ecx(source_info, |this| {
+                        this.ecx.read_immediate(this.ecx.eval_operand(op, None)?)
+                    })?;
+
+                    // Do not try to read bits for ZSTs
```

Comment:
is this reachable? I mean `ty.is_integral()` already checks whether we have an integer like type and these can't be zst, right?

Reply:
Strangely yes, without this, https://github.com/rust-lang/rust/blob/a916ac22b9f7f1f0f7aba0a41a789b3ecd765018/src/test/ui/consts/const-nullary-univariant-enum.rs ICEs with a "don't try to read ZST bits" message.

Reply:
Oh... right... Please leave a comment that this only exists for the "zst enum variant still has a discriminant that can be cast to an int" hack.